src/libraries/System.Private.CoreLib/src/System/Guid.cs

```diff
@@ -32,6 +34,19 @@ public partial struct Guid : IFormattable, IComparable, IComparable<Guid>, IEqua
         private byte _j;  // Do not rename (binary serialization)
         private byte _k;  // Do not rename (binary serialization)
 
+        private static uint* GetTableToHex()
```

Comment:
Is `Guid.ToString` so perf-critical that we're willing to burn 1KB of memory in the static ctor?

Reply:
I think that table can be moved into and used for "free" (just ptr+offset) conversion from bytes to hex chars https://github.com/dotnet/runtime/blob/master/src/libraries/Common/src/System/HexConverter.cs